### ISSUE TITLE
Property accessors trip up `no-unneeded-ternary/defaultAssignment`

### ISSUE THREAD

**Issue by garborg** [eslint/eslint#4654]
eslint version:
```sh
eslint -v
v1.10.3
```

test.js
```js
/*eslint no-unneeded-ternary: [2, {"defaultAssignment": false}]*/

var a = { b: { c: 1 } };
var foo = a ? a : 1;
var baz = a.b.c ? a.b.c : 0;
```

eslint output
```sh
eslint test.js

/Users/sean/Desktop/test.js
  4:15  error  Unnecessary use of conditional expression for default assignment  no-unneeded-ternary
```

So `a ? a :` is caught, but `a.b.c ? a.b.c :` is not.

**Comment by eslintbot** [comment]
@garborg Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Verified in ESLint demo.

**Comment by michaelficarra** [comment]
Why should this be reported? That expression can have arbitrary effects.

**Comment by garborg** [comment]
Not in practice here, so catching would be more useful for me, but I understand if creating false positives and/or complicating the optional configuration are all ruled out as fixes.

**Comment by michaelficarra** [comment]
For expressions containing only simple member accesses, I wouldn't be opposed. If you have side effecting getters, you deserve what you get.

**Comment by nzakas** [comment]
Yeah, I'm not 100% convinced on this. Properties can have side effects, and I'm pretty sure that's why we left this check off to begin with.

**Comment by nzakas** [comment]
No consensus here, so closing.
